### ISSUE TITLE
`update`/`capture` fails to interpret paths while using path variables

### ISSUE THREAD

**Issue by gilgardosh** [opticdev/optic#2036]
**Describe the bug**
Schema:
```
{
  "servers": [
    {
      "url": "https://www.host.com/api/users"
    }
  ],
  "paths": {
    "/{version}/name": {
      ...
    },
    "/{version}/email": {
      ...
    },
    "/v2/username": {
      ...
    },
    "/v2/pass": {
      ...
    },
    "/v1/username": {
      ...
    },
    "/v1/password": {
      ...
    }
  }
}
```

Operation URL: `https://www.host.com/api/users/v2/username`

Executing `update` on this path results in new schema path:
```
  "{version}/v2/username": {
    ...
  }
```

**Expected behavior**
For the patch to be merged with `"v2/username"` path

**Details (please complete the following information):**
 - OS and version: Mac OS 13.2.1
 - Optic version: v0.45.4
 - NodeJS version: 18.13.0

**Comment by niclim** [comment]
Hi, just an update here - I'm working on a fix in https://github.com/opticdev/optic/pull/2043, just as a summary of what's happening:
- We had a bug with prefixed urls (e.g. `https://example.com/api/users`) that I'm fixing in the above PR
- If you have ambiguous paths, you can also manually specify the path it should be. For example:
  - `/api/users/123456` might be documented as `/api/users/123456`  (optic tries to guess, but sometimes it needs help). to fix this case, you can add a path `/api/users/{userId}` to the `paths` key
  - We're working on redesigning the capture flow so it's more intuitive to use and handle ambiguous path patterns

I've got a prerelease build of this if you want to try this out now `0.47.7-0` - you can install this by running `npm i -g @useoptic/optic@0.47.7-0`. We're looking to release this next week and I'll report back here when there's a new version.

**Comment by niclim** [comment]
Released version `0.47.7` which includes the above fix
